pkg/common/oras/authprovider/alibabacloud/alibabacloudacrbasic.go

```diff
@@ -0,0 +1,232 @@
+	getAuthorizationTokenRequest.InstanceId = tea.String(instanceID)
+	runtime := &util.RuntimeOptions{}
+
+	tokenResponse, err := crClient.GetAuthorizationTokenWithOptions(getAuthorizationTokenRequest, runtime)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get acr authorization token: %w", err)
+	}
+	return tokenResponse.Body, nil
+}
+
+// Create returns an Alibaba CloudAcrBasicProvider
+func (d *AlibabaCloudAcrBasicProviderFactory) Create(authProviderConfig provider.AuthProviderConfig) (provider.AuthProvider, error) {
+	conf := alibabacloudAcrBasicAuthProviderConf{}
+	authProviderConfigBytes, err := json.Marshal(authProviderConfig)
+	if err != nil {
+		return nil, re.ErrorCodeConfigInvalid.WithError(err).WithComponentType(re.AuthProvider)
+	}
+
+	if err = json.Unmarshal(authProviderConfigBytes, &conf); err != nil {
+		return nil, re.ErrorCodeConfigInvalid.NewError(re.AuthProvider, alibabacloudAcrAuthProviderName, re.AuthProviderLink, err, "failed to parse alibaba cloud auth provider configuration", re.HideStackTrace)
+	}
+	var acrInstancesConfig = make(map[string]string, 0)
+	if len(conf.AcrInstancesConfig) > 0 {
+		for _, ic := range conf.AcrInstancesConfig {
+			acrInstancesConfig[ic.InstanceName] = ic.InstanceID
+		}
+	}
+	//get ACR EE instance id
+	instanceID := conf.DefaultInstanceID
+	if instanceID == "" {
+		instanceID = os.Getenv("ALIBABA_CLOUD_ACR_INSTANCE_ID")
+		if instanceID == "" && len(acrInstancesConfig) == 0 {
+			return nil, re.ErrorCodeEnvNotSet.WithComponentType(re.AuthProvider).WithDetail("no instance ID provided and ALIBABA_CLOUD_ACR_INSTANCE_ID environment variable is empty")
+		}
+	}
+	acrInstancesConfig[defaultInstance] = instanceID
+	acrAuthToken := AcrAuthToken{}
+	acrAuthToken.AuthData = make(map[string]*cr20181201.GetAuthorizationTokenResponseBody)
+
+	return &alibabacloudAcrBasicAuthProvider{
+		acrAuthToken:       acrAuthToken,
+		acrInstancesConfig: acrInstancesConfig,
+		providerName:       alibabacloudAcrAuthProviderName,
+		getAcrAuthToken:    &defaultAcrAuthTokenGetterImpl{},
+	}, nil
+}
+
+// Enabled checks for non-empty AlibabaCloud RAM creds
+func (d *alibabacloudAcrBasicAuthProvider) Enabled(_ context.Context) bool {
+	if d.providerName == "" {
+		logrus.Error("basic Alibaba Cloud ACR providerName was empty")
+		return false
+	}
+
+	return true
+}
+
+// Provide returns the credentials for a specified artifact.
+// Uses AlibabaCloud RRSA to retrieve creds from RRSA credential chain
+func (d *alibabacloudAcrBasicAuthProvider) Provide(ctx context.Context, artifact string) (provider.AuthConfig, error) {
+	if !d.Enabled(ctx) {
+		return provider.AuthConfig{}, fmt.Errorf("Alibaba Cloud RRSA basic auth provider is not properly enabled")
+	}
+
+	registry, err := provider.GetRegistryHostName(artifact)
+	if err != nil {
+		return provider.AuthConfig{}, errors.Wrapf(err, "could not get ACR registry from %s", artifact)
+	}
+
+	if !d.acrAuthToken.exists(registry) {
```

Comment:
another scenario to validate if this is applicable :   1. get a token where the token does not have proper access to the registry . 2. Validate remote call to registry fails . 3 assign the correct permission and ensure artifact can be fetched from registry